diff --git a/shell/90_aliases.sh b/shell/90_aliases.sh
index 8443beb..7f697be 100644
--- a/shell/90_aliases.sh
+++ b/shell/90_aliases.sh
@@ -3,7 +3,7 @@
 #
 
 has_command ip && default_iface=$(ip -o -4 route show to default | awk '{print $5}')
-has_command route && route get default > /dev/null && default_iface=$(route get default | grep interface: | awk '{print $2}')
+has_command route && route get default >/dev/null 2>&1 && default_iface=$(route get default | grep interface: | awk '{print $2}')
 
 if ls --color=auto -d . >/dev/null 2>&1; then
   # echo "has GNU ls"
@@ -80,6 +80,11 @@ alias map="xargs -n1"
 # (useful when executing time-consuming commands)
 alias bell="tput bel"
 
+alias home="cd $HOME"
+alias dls="cd $HOME/Downloads"
+alias doc="cd $HOME/Documents"
+alias path="echo $PATH | sed -e 's/:/\n/g'"
+
 if [[ "$OSTYPE" == "darwin"* ]]; then
 
   if [[ -x /Applications/MacVim.app/Contents/MacOS/Vim ]]; then
diff --git a/shell/bash/50_title.sh b/shell/bash/50_title.sh
index ef46cef..c3eaaf4 100644
--- a/shell/bash/50_title.sh
+++ b/shell/bash/50_title.sh
@@ -1,8 +1,8 @@
 case $TERM in
-    xterm*|rxvt*|Eterm|aterm|kterm|gnome)
+    xterm*|rxvt*|Eterm*|aterm*|kterm*|gnome*)
         PROMPT_COMMAND="$PROMPT_COMMAND"$'\n''echo -ne "\033]0;${PWD/$HOME/~} [${USER}@${HOSTNAME}]\007"'
         ;;
-    screen)
+    screen*|tmux*)
         PROMPT_COMMAND="$PROMPT_COMMAND"$'\n''echo -ne "\033_${PWD/$HOME/~}\033\\"'
         ;;
     linux*)
diff --git a/shell/zsh/50_completion.zsh b/shell/zsh/50_completion.zsh
index b578707..8789224 100644
--- a/shell/zsh/50_completion.zsh
+++ b/shell/zsh/50_completion.zsh
@@ -4,6 +4,7 @@ setopt complete_in_word # starting completion doesn't move cursor. complete from
 setopt always_to_end # Move cursor to the end of a completed word.
 
 # add additional zsh completions
+[ -d /usr/share/zsh/vendor-completions ] && fpath=(/usr/share/zsh/vendor-completions $fpath)
 fpath=($DOTFILES/thirdparty/zsh-completions/src $fpath)
 
 # completion context string has a set of fields
diff --git a/shell/zsh/50_title.zsh b/shell/zsh/50_title.zsh
index 9d32932..31196d3 100644
--- a/shell/zsh/50_title.zsh
+++ b/shell/zsh/50_title.zsh
@@ -1,12 +1,12 @@
 # xterm and rxvt title setting
 case $TERM in
-    xterm*|rxvt*|Eterm|aterm|kterm|gnome)
+    xterm*|rxvt*|Eterm*|aterm*|kterm*|gnome*)
         precmd() {
             # OK, this needs some documentaions, ]0 chooses title and icon title
             echo -ne "\033]0;${PWD/$HOME/~} [${USER}@${HOST}]\007"
         }
         ;;
-    screen)
+    screen*|tmux*)
         precmd() {
             # this clears the program running status and reset it to pwd
             echo -ne "\033kzsh\033\\" # clear title string
diff --git a/shell/zsh/90_options.zsh b/shell/zsh/90_options.zsh
index 3d2675f..3bc4bc7 100644
--- a/shell/zsh/90_options.zsh
+++ b/shell/zsh/90_options.zsh
@@ -30,6 +30,9 @@ bindkey -v
 # list key maps in order of emacs vicmd viins like in the man page
 bind2maps             viins -- 'jj' vi-cmd-mode
 
+# allow deleting past vi insert mode start point
+bind2maps             viins -- '^?' backward-delete-char
+
 bind2maps emacs       viins -- ' ' magic-space
 
 bind2maps emacs       viins -- '^[[Z' reverse-menu-complete
